src/main/java/org/elasticsearch/search/highlight/HighlightPhase.java

```diff
@@ -76,12 +75,12 @@ public boolean hitExecutionNeeded(SearchContext context) {
     public void hitExecute(SearchContext context, HitContext hitContext) {
         Map<String, HighlightField> highlightFields = newHashMap();
         for (SearchContextHighlight.Field field : context.highlight().fields()) {
-            List<String> fieldNamesToHighlight;
+            Set<String> fieldNamesToHighlight;
```

Comment:
I'm thinking something like:
```json
"highlight": {
  "cat_*": <other stuff>,
  "*": <stuff>
}
```

As is now the last matching regex wins but it does so by highlighting the field twice. That's what I meant.